### PR TITLE
Added definitions for react-dynamic-number npm package 

### PR DESCRIPTION
**Pull request by eugrdn**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-dynamic-number/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import * as React from 'react';
+
+/**
+ * remove Diff & Omit when in will be placed in TS from scratch
+ */
+type Diff<T extends string, U extends string> = ({[P in T]: P} & {[P in U]: never} & {[x: string]: never})[T];
+type Omit<T, K extends keyof T> = {[P in Diff<keyof T, K>]: T[P]};
+
+interface OmittedFields {
```
Comment (andy-ms) [inline types/react-dynamic-number/index.d.ts]: This is only used for its keys -- just use `"ref" | "value" | "onChange" | "placeholder"` instead

File: types/react-dynamic-number/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  onChange: any;
+  placeholder: any;
+}
+
+type BaseInputProps = Partial<
+  Omit<React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>, keyof OmittedFields>
+>;
+
+type Value = number | '';
+type Separator = '.' | ',';
```
Comment (andy-ms) [inline types/react-dynamic-number/index.d.ts]: Type is only used once -- just inline (`separator?: '.' | ','`). Same for the other two.

File: types/react-dynamic-number/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  integer?: number;
+  fraction?: number;
+  positive?: boolean;
+  negative?: boolean;
+  placeholder?: string;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>, modelValue: number, viewValue: string) => void;
+}
+
+declare class DynamicNumber extends React.Component<DynamicNumberProps> {}
+export default DynamicNumber;
```
Comment (andy-ms) [inline types/react-dynamic-number/index.d.ts]: `export default class DynamicNumber extends React.Component<DynamicNumberProps> {}` should work.

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/react-dynamic-number/index.d.ts (hunk 3 above)
Comment (eugrdn) [reply types/react-dynamic-number/index.d.ts]: Is that a common code style requirement of the DefinitelyTyped or just your opinion?

Comment (andy-ms) [reply types/react-dynamic-number/index.d.ts]: We don't have a lint rule for it, but I think we would if we had the time. No need to declare something and then export it if you can export directly.

Comment (eugrdn) [reply types/react-dynamic-number/index.d.ts]: Ok. So, if I remove the `declare` statement there are some tslint errors appearing for the types that not needed to be exported (like Omit or BaseInputProps): "Prefer to explicitly write 'export' for an external module export."

Comment (andy-ms) [reply types/react-dynamic-number/index.d.ts]: Just change that to `export type Diff = ...`. A declaration in a `.d.ts` module declaration is always exported whether you write `export` or not, so the lint rule is just asking to make it explicit.

File: types/react-dynamic-number/index.d.ts (hunk 1 above)
Comment (eugrdn) [reply types/react-dynamic-number/index.d.ts]: done

File: types/react-dynamic-number/index.d.ts (hunk 2 above)
Comment (eugrdn) [reply types/react-dynamic-number/index.d.ts]: done

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**